### ISSUE TITLE
Support for Kramer VS-41H HDMI switcher

### ISSUE THREAD

**Issue by chrisrouse** [bitfocus/companion#555]
This request is coming via the ProPresenter forums. One of our users requested support for TCP, but it would be a long time coming, if ever. Since it's possible to trigger Companion indirectly with MIDI, I'd rather see the module here and then help him control Companion.

He provided a link to a Python script that works: https://github.com/UbiCastTeam/kramer-python-control/blob/master/kramer_switch.py

This switcher uses the Protocol 2000. https://k.kramerav.com/downloads/protocols/protocol_2000_rev0_51.pdf

**Comment by SteveTherrien** [comment]
I threw together something based on my Kramer Matrix module (it uses the same protocol as the VS-41H switcher). The new module can be found at https://github.com/SteveTherrien/companion-module-kramer-vs41h

I don't have access to a VS-41H switcher, but a packet capture shows the module outputting the correct instructions. It should work.

Is it possible for the requester to run Companion from source and manually add this module so he can give it a try?

**Comment by chrisrouse** [comment]
Yeah, I’ll try to reach out to him this week. Can he download the latest stable and add this or does he need to setup the full test environment?

**Comment by SteveTherrien** [comment]
No, it'd have to be a test environment, unless someone can create a custom Companion build for his OS that contains this module.

**Comment by JeffreyDavidsz** [comment]
We can create a custom build with that module and remove it after. There is a way to make a build offline also of course

**Comment by chrisrouse** [comment]
I’ll see if that guy is comfortable setting it up manually to test.

Thanks,
Chris Rouse

**Comment by chrisrouse** [comment]
I never got any reply from the guy in the forums. I guess he didn't turn on notifications for his post.
